src/test/kotlin/dev/stalla/RealLifeFeedsTest.kt

```diff
@@ -5,40 +5,35 @@ import assertk.assertThat
 import assertk.assertions.exists
 import assertk.assertions.isEqualTo
 import assertk.assertions.isNotNull
-import org.junit.jupiter.api.DynamicTest
-import org.junit.jupiter.api.TestFactory
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
 import java.io.File
 
+@ExtendWith(TemporaryFileParameterResolver::class)
 internal class RealLifeFeedsTest {
 
-    private val resourceFilesPath = "/xml/real-life-feeds"
+    private class RealLifeFeedsProvider : ArgumentsProvider by arguments(*allResourceFilesIn(resourceFilesPath).toTypedArray())
 
-    @TestFactory
-    fun dynamicTestsWithCollection(): Collection<DynamicTest> {
-        val testFiles = allResourceFilesIn(resourceFilesPath)
-
-        return testFiles.map { testFile ->
-            DynamicTest.dynamicTest(
-                "[${testFile.nameWithoutExtension}] should be able to parse, write and re-parse a real-life feed correctly"
-            ) { testRealLifeFeed(testFile) }
-        }
-    }
-
-    private fun testRealLifeFeed(file: File) {
+    @ParameterizedTest
```

Comment:
I'd add a `name = "Can parse, write and re-parse a real-life feed: {0}"` for ease of use